core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -118,18 +141,37 @@ class TaskMetrics extends Serializable {
     _inputMetrics = inputMetrics
   }
 
+  private var _outputMetrics: Option[OutputMetrics] = None
+
   /**
-   * If this task writes data externally (e.g. to a distributed filesystem), metrics on how much
-   * data was written are stored here.
+   * Metrics related to writing data externally (e.g. to a distributed filesystem),
+   * defined only in tasks with output.
    */
-  var outputMetrics: Option[OutputMetrics] = None
+  def outputMetrics: Option[OutputMetrics] = _outputMetrics
+
+  @deprecated("setting OutputMetrics is for internal use only", "2.0.0")
```

Comment:
Do you know whether there's any third-party code that calls this? Wondering if we can just drop it. Also, AFAIK this would only be for source-compatibility: code which directly set `outputMetrics` will be binary-incompatible with this.